src/plugins/data/server/ui_settings.ts

```diff
@@ -0,0 +1,643 @@
+          'This configuration enhances support for dashboards containing visualizations accessing dissimilar indexes. ' +
+          'When disabled, all filters are applied to all visualizations. ' +
+          'When enabled, filter(s) will be ignored for a visualization ' +
+          `when the visualization's index does not contain the filtering field.`,
+      }),
+      category: ['search'],
+      schema: schema.boolean(),
+    },
+    'courier:setRequestPreference': {
+      name: i18n.translate('kbn.advancedSettings.courier.requestPreferenceTitle', {
+        defaultMessage: 'Request preference',
+      }),
+      value: 'sessionId',
+      options: ['sessionId', 'custom', 'none'],
+      optionLabels: requestPreferenceOptionLabels,
+      type: 'select',
+      description: i18n.translate('kbn.advancedSettings.courier.requestPreferenceText', {
+        defaultMessage: `Allows you to set which shards handle your search requests.
+          <ul>
+            <li><strong>{sessionId}:</strong> restricts operations to execute all search requests on the same shards.
+              This has the benefit of reusing shard caches across requests.</li>
+            <li><strong>{custom}:</strong> allows you to define a your own preference.
+              Use <strong>courier:customRequestPreference</strong> to customize your preference value.</li>
+            <li><strong>{none}:</strong> means do not set a preference.
+              This might provide better performance because requests can be spread across all shard copies.
+              However, results might be inconsistent because different shards might be in different refresh states.</li>
+          </ul>`,
+        values: {
+          sessionId: requestPreferenceOptionLabels.sessionId,
+          custom: requestPreferenceOptionLabels.custom,
+          none: requestPreferenceOptionLabels.none,
+        },
+      }),
+      category: ['search'],
+      schema: schema.string(),
+    },
+    'courier:customRequestPreference': {
+      name: i18n.translate('kbn.advancedSettings.courier.customRequestPreferenceTitle', {
+        defaultMessage: 'Custom request preference',
+      }),
+      value: '_local',
+      type: 'string',
+      description: i18n.translate('kbn.advancedSettings.courier.customRequestPreferenceText', {
+        defaultMessage:
+          '{requestPreferenceLink} used when {setRequestReferenceSetting} is set to {customSettingValue}.',
+        description:
+          'Part of composite text: kbn.advancedSettings.courier.customRequestPreference.requestPreferenceLinkText + ' +
+          'kbn.advancedSettings.courier.customRequestPreferenceText',
+        values: {
+          setRequestReferenceSetting: '<strong>courier:setRequestPreference</strong>',
+          customSettingValue: '"custom"',
+          requestPreferenceLink:
+            '<a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-preference.html" target="_blank" rel="noopener">' +
```

Comment:
@lukeelmers not sure that it's related to current PR, but it looks like an issue which should be addressed in future. 
We should get links from `DocLinks` service (src/core/public/doc_links/doc_links_service.ts) instead of hardcoding it.

Reply:
I created https://github.com/elastic/kibana/issues/67026 to track updating docLinks as a follow-up task.